pysmt/optimization/z3.py

```diff
@@ -42,7 +42,12 @@ def __init__(self, environment, logic, **options):
         self.z3 = z3.Optimize()
 
     def _assert_z3_goal(self, goal):
-        obj = self.converter.convert(goal.term())
+        term = goal.term()
+        ty = self.environment.stc.get_type(term)
+        if goal.signed() and ty.is_bv_type():
```

Comment:
Maybe add a comment with a link to the Z3 issue discussing this: https://github.com/Z3Prover/z3/issues/1339